tests/components/abode/test_init.py

```diff
@@ -0,0 +1,63 @@
+"""Tests for the Abode module."""
+from abodepy.exceptions import AbodeException
+import abodepy.helpers.constants as CONST
+from abodepy.helpers.errors import INVALID_SETTING
+from requests.exceptions import ConnectTimeout
+
+from homeassistant.components import abode
+from homeassistant.components.alarm_control_panel import DOMAIN as ALARM_DOMAIN
+
+from .common import setup_platform
+
+
+async def test_abode_setup_from_config(hass, requests_mock):
+    """Test setup from configuration yaml file."""
+    config = {"abode": {"username": "foo", "password": "bar", "polling": True}}
+    response = await abode.async_setup(hass, config)
+    assert response
+
+
+async def test_connection_error(hass, requests_mock):
+    """Test exception when unable to connect to Abode."""
+    requests_mock.post(CONST.LOGIN_URL, exc=ConnectTimeout)
+    # Tried using 'with pytest.raises(ConnectTimeout):' but
```

Comment:
For whatever reason, `ConnectTimeout` is not being raised. The strange part is I know the `request_mock.post(CONST.LOGIN_URL, exc=ConnectTImeout)` line is being used because when I run the test, I can see it's executing the appropriate part of the abode integration and logging the correct message.  However, where it prints out the exception in that log message is blank.